src/classes/BDI_HelpTextCMT.cls

```diff
@@ -0,0 +1,249 @@
+                        this.importStatusFieldName = field;
+                    } else {
+                        this.importLookupFieldName = field;
+                    }
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Wrapper class for help text field
+     */
+    public class HelpTextParts {
+        public String objectName;
+        public String fieldName;
+        public String objectAPIName;
+
+        public HelpTextParts(String helpText) {
+            List<String> parts = helpText.split('\\.', 2);
+
+            this.objectName = String.isNotBlank(parts[0]) ? parts[0] : null;
+            this.fieldName = String.isNotBlank(parts[1]) ? parts[1] : null;
+
+            if (String.isNotBlank(this.objectName)) {
+                this.objectAPIName = DI_OBJECT_MAP.get(this.objectName);
+            }
+        }
+
+    }
+
+    /**
+     * Intermediary wrapper class for Data_Import_Field_Mapping__mdt; Probably scrub later and use DataImportFieldMapping directly.
+     */
+    public class MappingDetail {
+        public String dataImportObjectName;
+        public String targetObjectName;
+        public String dataImportFieldLabel;
+        public String targetFieldAPIName;
+        public String importedRecordField;
+        public String importedRecordStatusField;
+
+        public MappingDetail(String dataImportObjectName, String dataImportFieldLabel, String targetObjectName,
+                             String targetFieldAPIName, String importedRecordField, String importedRecordStatusField) {
+            this.dataImportObjectName = dataImportObjectName;
+            this.dataImportFieldLabel = dataImportFieldLabel;
+            this.targetObjectName = targetObjectName;
+            this.targetFieldAPIName = targetFieldAPIName;
+            this.importedRecordField = importedRecordField;
+            this.importedRecordStatusField = importedRecordStatusField;
+        }
+
+    }
+
+    /**
+     * @description Utility to convert a name into a valid 'DeveloperName' quality value (or unique GUID)
+     * for the CMT record
+     * @param name String
+     * @param makeUnique Boolean
+     * @return DeveloperName format string
+     */
+    private static String generateRecordName(String name, Boolean makeUnique, String objectLabel, String objectAPIName) {
+        name = name.replaceAll('[^\\w]+', '_').replaceAll('_{2,}', '_');
+
+        if (String.isNotBlank(objectLabel)) {
+            objectLabel = objectLabel.replace(' ', '_');
+            name = name.replace(objectAPIName, objectLabel);
+        }
+
+        if (name.length() > 30) {
+            String firstFifteen = name.substring(0, 15);
+            String lastFifteen = name.substring(name.length() - 15, name.length());
+            name = firstFifteen + lastFifteen;
+        }
+
+        if (makeUnique == true) {
+            // Create a GUID format unique string
+            Blob b = Crypto.GenerateAESKey(128);
```

Comment:
Is the key generated by GenerateAESKey() guaranteed unique? Or do we run the risk of incredibly-unlikely duplicates?